lib/chef/provider/aws_cache_cluster.rb

```diff
@@ -0,0 +1,65 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsCacheCluster < Chef::Provisioning::AWSDriver::AWSProvider
+
+  protected
+
+  def create_aws_object
+    converge_by "create new Elasticache Cluster #{new_resource.name} in #{region}" do
+      new_resource.driver.elasticache.create_cache_cluster(desired_options)
+    end
+  end
+
+  def update_aws_object(cache_cluster)
+    if update_required?(cache_cluster)
+      converge_by "update Elasticache Cluster #{new_resource.name} in #{region}" do
+        new_resource.driver.elasticache.modify_cache_cluster(desired_options)
+      end
+    end
+  end
+
+  def destroy_aws_object(cache_cluster)
+    converge_by "delete Elasticache Cluster #{new_resource.name} in #{region}" do
+      new_resource.driver.elasticache.delete_cache_cluster(
+        cache_cluster_id: cache_cluster[:cache_cluster_id]
+      )
+    end
+  end
+
+  private
+
+  # def update_cache_subnet_group
+  #   new_resource.driver.elasticache.modify_cache_subnet_group(desired_options)
+  # end
+
+  def desired_options
+    @desired_options ||= begin
+      options = {}
+      options[:cache_cluster_id] = new_resource.cluster_name
```

Comment:
Yes, in this case the 'id' is a friendly name instead of the normal aws `sg-1234abcd`. I could change the attribute to be `cluster_id` but `cluster_name` felt more indicative of what it can accept.